### PR TITLE
fix: delay rendering slider internals until after direction check

### PR DESCRIPTION
**Pull request by scomea**
# Description
The slider component can only query the dom for direction until after it has mounted.  Previously we rendered in default ltr and if we detect rtl after the component mounted we reset the state and triggered a render to correct the layout to rtl.  This meant we ran layout logic twice in rtl and that css transitions could be triggered by the ltr to rtl layout changes.  With this change we only render the component root element on the first pass, check for direction on componentDidMount() and only render the component internals at that point.

## Motivation & context
Users reported seeing transition animations on rtl ui.  Slight perf gain on rtl by avoiding running layout twice.

## Issue type checklist
- [ ] **Chore**: A change that does not impact distributed packages.
- [x] **Bug fix**: A change that fixes an issue, link to the issue above.
- [ ] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

## Process & policy checklist
- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/en/contributing/standards) for this project.

### CODE REVIEW

File: packages/fast-components-react-base/src/slider/slider.tsx [hunk 1]
```diff
@@ -174,6 +175,7 @@ class Slider extends Foundation<SliderHandledProps, SliderUnhandledProps, Slider
             incrementDirection: 1,
             usePageStep: false,
             direction: Direction.ltr,
```
Comment (nicholasrice) [inline packages/fast-components-react-base/src/slider/slider.tsx]: I think you can avoid adding an additional state property by defaulting the `direction` state property to `null`. Then the check in render becomes "do I know my direction? If not, don't render anything", which feels more in line with the purpose of this change.

Comment (chrisdholt) [reply packages/fast-components-react-base/src/slider/slider.tsx]: 🔥 🔥 🔥 - I like this.

Comment (scomea) [reply packages/fast-components-react-base/src/slider/slider.tsx]: yup, good call

File: packages/fast-components-react-base/src/slider/slider.spec.tsx [hunk 2]
```diff
@@ -767,6 +768,21 @@ describe("Slider", (): void => {
...
+        const rendered: any = mount(
+            <Slider managedClasses={managedClasses} initialValue={50} />,
+            {
+                attachTo: container,
+            }
+        );
+
+        expect(rendered.state("direction")).toBe(Direction.ltr);
+        document.body.removeChild(container);
+    });
```
Comment (chrisdholt) [inline packages/fast-components-react-base/src/slider/slider.spec.tsx]: @scomea should we add a test to ensure that if the direction is null the slider does not render?

Comment (scomea) [reply packages/fast-components-react-base/src/slider/slider.spec.tsx]: figured one out
